src/snowflake/snowpark/_internal/type_utils.py

```diff
@@ -655,14 +662,15 @@ def retrieve_func_type_hints_from_source(
     Returns None if the function is not found.
     """
 
-    def parse_arg_annotation(annotation: ast.expr) -> str:
+    def parse_arg_annotation(annotation: Union[ast.expr, ast.slice]) -> str:
         if isinstance(annotation, (ast.Tuple, ast.List)):
             return ", ".join([parse_arg_annotation(e) for e in annotation.elts])
         if isinstance(annotation, ast.Attribute):
             return f"{parse_arg_annotation(annotation.value)}.{annotation.attr}"
         if isinstance(annotation, ast.Subscript):
             return f"{parse_arg_annotation(annotation.value)}[{parse_arg_annotation(annotation.slice)}]"
         if isinstance(annotation, ast.Index):
```

Comment:
how about changing this if to `if isinstance(annotation, ast.Index) and hasattr(annotation, "value")`?